### ISSUE TITLE
DDLDatabase fails with MySQL TINYINT(1) type columns

### ISSUE THREAD

**Issue by anuraaga** [jOOQ/jOOQ#7020]
Parser doesn't seem to recognize TINYINT(1) (boolean)

JOOQ 3.10.3

```
org.jooq.impl.ParserException: C42000_NO_SUBCLASS: Expected tokens: [AUTOINCREMENT, AUTO_INCREMENT, CHECK, DEFAULT, GENERATED, NOT NULL, NULL, PRIMARY KEY, UNIQUE]: plink_frequencies BLOB NOT NULL,
  is_test TINYINT[*](1) DEFAULT '0' NOT NULL,
  updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT N
        at org.jooq.impl.ParserImpl$ParserContext.unexpectedToken(ParserImpl.java:5791)
        at org.jooq.impl.ParserImpl.parseQuery(ParserImpl.java:547)
        at org.jooq.impl.ParserImpl.parse(ParserImpl.java:345)
        at org.jooq.impl.ParserImpl.parse(ParserImpl.java:337)
        at org.jooq.util.ddl.DDLDatabase.create0(DDLDatabase.java:124)
        at org.jooq.util.AbstractDatabase.create(AbstractDatabase.java:237)
        at org.jooq.util.AbstractDatabase.create(AbstractDatabase.java:226)
        at org.jooq.util.AbstractDatabase.getDialect(AbstractDatabase.java:208)
        at org.jooq.util.AbstractGenerator.logDatabaseParameters(AbstractGenerator.java:131)
        at org.jooq.util.JavaGenerator.generate(JavaGenerator.java:271)
        at org.jooq.util.GenerationTool.run(GenerationTool.java:642)
        at org.jooq.util.GenerationTool.generate(GenerationTool.java:203)
        at org.jooq.util.GenerationTool.main(GenerationTool.java:175)
```

**Comment by lukaseder** [comment]
Thanks for reporting. Indeed, those integer lengths can be parsed, and safely ignored as they have no meaning in MySQL (other than hinting at a display length, and indicating the intent of the developer to use a type as a boolean).

**Comment by lukaseder** [comment]
This is now fixed for jOOQ 3.11 and will be merged to 3.10.4 next week (#7030)
